support-frontend/assets/helpers/page/commonReducer.js

```diff
@@ -56,6 +60,43 @@ function createCommonReducer(initialState: CommonState): (state?: CommonState, a
           },
         };
 
+      case 'SET_CURRENCY_ID':
+        return {
+          ...state,
+          internationalisation: {
+            ...state.internationalisation,
+            currencyId: (action.useLocalCurrency && state.internationalisation.localCurrencyCountry)
+              ? state.internationalisation.localCurrencyCountry.currency
+              : state.internationalisation.defaultCurrency,
+          },
+        };
+
+      case 'SET_LOCAL_CURRENCY_COUNTRY':
+        return {
+          ...state,
+          internationalisation: {
+            ...state.internationalisation,
+            localCurrencyCountry: action.localCurrencyCountry,
+          },
+        };
+
+      case 'SET_USE_LOCAL_CURRENCY_FLAG':
```

Comment:
Can we achieve the same with only this action? It can set `currencyId` and `amounts` based on `action.useLocalCurrency`

Reply:
discussed IRL:
we need to store whether or not the user has asked to see local currency (for switching between contribution types).
And we need to update amounts + currency based on the contributionType + useLocalCurrency

So we'll keep them as separate actions

Reply:
👍🏼